_posts/2025-04-28-sparse-autodiff.md

```diff
@@ -383,11 +387,17 @@ We can then materialize multiple rows in a single VJP.
 ### Sparsity pattern detection and coloring
 
 Unfortunately, our initial assumption had a major flaw: 
-Since AD only gives us a composition of linear maps and linear maps are black-box functions,
+since AD only gives us a composition of linear maps and linear maps are black-box functions,
 the structure of the Jacobian is completely unknown.
+In other words, **we cannot tell which rows and columns are orthogonal without first materializing a Jacobian matrix.**
+But if we fully materialize a Jacobian via traditional AD, ASD isn't needed at all.
 
-**We can't tell which rows and columns are orthogonal without first materializing a Jacobian matrix.**
-But if we fully materialize a Jacobian via traditional AD, ASD isn't needed.
+The solution to this problem is shown in Figure 10 (a):
+in order to find orthogonal columns (or rows), we don't need to materialize the full Jacobian.
+Instead, it is enough to **detect the sparsity pattern** of the Jacobian.
+This binary-valued pattern contains enough information to deduce orthogonality.
+From there, we use a **coloring algorithm** to group mutually orthogonal columns (or rows) together.
+Such a coloring can be visualized on Figure 10 (b), where the yellow columns will be evaluated together and the light blue ones too.
```

Comment:
```suggestion
Such a coloring can be visualized on Figure 10 (b), where the yellow columns will be evaluated together (first JVP) and the light blue ones will be evaluated together (second JVP), for a total of 2 JVPs instead of 5.
```